wagtail/blocks/base.py

```diff
@@ -564,14 +564,11 @@ def render_with_errors(self, name, value, attrs=None, errors=None, renderer=None
             error = errors.as_data()[0]
             error_json = json.dumps(get_error_json_data(error))
         else:
-            error_json = "null"
+            error_json = json.dumps({})
 
         return format_html(
             """
-                <div id="{id}" data-block="{block_json}" data-value="{value_json}" data-error="{error_json}"></div>
-                <script>
-                    initBlockWidget('{id}');
-                </script>
+                <div id="{id}" data-block data-controller="w-block" data-w-block-data-value="{block_json}" data-w-block-initial-value="{value_json}" data-w-block-error-value="{error_json}"></div>
```

Comment:
```suggestion
                <div id="{id}" data-block data-controller="w-block" data-w-block-data-value="{block_json}" data-w-block-arguments-value="[{value_json},{error_json}]"></div>
```
Finally, we would provide a single `arguments` value that is an array `[...,...]` value with the two JSON strings.